std/haxe/ds/BalancedTree.hx

```diff
@@ -128,6 +128,12 @@ class BalancedTree<K,V> implements haxe.Constraints.IMap<K,V> {
 		keysLoop(root, ret);
 		return ret.iterator();
 	}
+	
+	public function copy():BalancedTree<K, V> {
+		var copied = new BalancedTree<K, V>();
+		for(key in keys()) copied.set(key, get(key));
```

Comment:
The way I see it tree nodes are never mutated, so doing `copied.root = this.root` should be possible.